backend/cmm_helpers.mli

```diff
@@ -1078,14 +1081,79 @@ val unboxed_vec128_array_length : expression -> Debuginfo.t -> expression
 val unboxed_float32_array_ref :
   expression -> expression -> Debuginfo.t -> expression
 
+(** Read an unboxed float32 from a 64-bit field in an array represented as
+    a mixed block (with tag zero), as used for unboxed product arrays.
+
+    The float32 is expected to be in the least significant bits of the
+    64-bit field.  The most significant 32 bits of such field are ignored.
+
+    The zero-indexed element number is specified as a tagged immediate.
+*)
+val unboxed_mutable_float32_unboxed_product_array_ref :
+  expression -> array_index:expression -> Debuginfo.t -> expression
+
+(** Write an unboxed float32 into a 64-bit field in an array represented as
+    a mixed block (with tag zero), as used for unboxed product arrays.
+
+    The zero-indexed element number is specified as a tagged immediate.
+
+    The float32 will be written to the least significant bits of the
+    64-bit field.  The top 32 bits of the written word will be initialized
+    to zero.  Note that two writes are involved.
```

Comment:
We could do `movss xmm xmm`, `movsd mem xmm` instead of separate writes

Reply:
Moved to a CR